### ISSUE TITLE
Failed to query history state data

### ISSUE THREAD

**Issue by hujw77** [darwinia-network/darwinia#862]
## Issue
I pick a xRING transfer tx test https://subview.xyz/tx/0xb61f1f51207a5819b26c444f54d7a4656f67431d88fe3ac60b82200dedbaf313/token-transfers

## block 10099222
xRING balance of 0x7379e6927ebece670cba86d31c9efb8cee11daed at block 10099222
```shell
+ curl -fsS https://crab-rpc.darwinia.network -d '{"id":1,"jsonrpc":"2.0","method":"eth_call","params":[{"data":"0x70a082310000000000000000000000007379e6927ebece670cba86d31c9efb8cee11daed","from":"0x7aE77149ed38c5dD313e9069d790Ce7085caf0A6","gas":"0x5b8d80","to":"0x7399Ea6C9d35124d893B8d9808930e9d3F211501"},"0x9a1a16"]}' -H 'Content-Type: application/json'
521308601529
```
## block 10099223
xRING balance of 0x7379e6927ebece670cba86d31c9efb8cee11daed at block 10099223
```shell
+ curl -fsS https://crab-rpc.darwinia.network -d '{"id":1,"jsonrpc":"2.0","method":"eth_call","params":[{"data":"0x70a082310000000000000000000000007379e6927ebece670cba86d31c9efb8cee11daed","from":"0x7aE77149ed38c5dD313e9069d790Ce7085caf0A6","gas":"0x5b8d80","to":"0x7399Ea6C9d35124d893B8d9808930e9d3F211501"},"0x9a1a17"]}' -H 'Content-Type: application/json'
521308601529
```
## block 10099224
xRING balance of 0x7379e6927ebece670cba86d31c9efb8cee11daed at block 10099224
```shell
+ curl -fsS https://crab-rpc.darwinia.network -d '{"id":1,"jsonrpc":"2.0","method":"eth_call","params":[{"data":"0x70a082310000000000000000000000007379e6927ebece670cba86d31c9efb8cee11daed","from":"0x7aE77149ed38c5dD313e9069d790Ce7085caf0A6","gas":"0x5b8d80","to":"0x7399Ea6C9d35124d893B8d9808930e9d3F211501"},"0x9a1a18"]}' -H 'Content-Type: application/json'
521308601529
```

**Comment by AsceticBear** [comment]
Are you sure the to `0x7399Ea6C9d35124d893B8d9808930e9d3F211501` is right? Why not `0x90e2fa4099eb1fed203769b99c9c64cdda1acb57`?
![image](https://user-images.githubusercontent.com/11801722/163954426-a4952329-d7da-4a03-ae83-f4c5ad0e08d3.png)

**Comment by hujw77** [comment]
xRING transfer flow: 
0x7379e6927EBECe670cbA86D31c9efb8CEE11dAed --> 0x90e2FA4099eb1fED203769B99c9c64cDDa1acB57 --> 0xA3304b1a0c88393e5d28Ded0B1E8C6d41f1B17d4

so the tx result is 0x7379e6927EBECe670cbA86D31c9efb8CEE11dAed -50 xRING
0xA3304b1a0c88393e5d28Ded0B1E8C6d41f1B17d4 +50 xRING

**Comment by hujw77** [comment]
`0x7399Ea6C9d35124d893B8d9808930e9d3F211501` is the xRING token address, 
the curl command means to get the xRING balance of `0x7379e6927EBECe670cbA86D31c9efb8CEE11dAed`

**Comment by AsceticBear** [comment]
@hujw77 The v0.12.0 crab RPC client doesn't support custom block_number to query history state data. Fortunately,  this had been fixed in the v0.12.1, which will release this week. Please test it again after the new release and RPC node upgrade. I'll let you know.

These two lines explain the reason https://github.com/darwinia-network/frontier/blob/darwinia-v0.12.0/client/rpc/src/eth.rs#L865-L866
